modules/sdk-coin-ada/src/lib/stakingWithdrawRewardsBuilder.ts

```diff
@@ -0,0 +1,44 @@
+import { BaseKey, TransactionType } from '@bitgo/sdk-core';
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { TransactionBuilder } from './transactionBuilder';
+import { Transaction } from './transaction';
+import * as CardanoWasm from '@emurgo/cardano-serialization-lib-nodejs';
+
+export class StakingWithdrawRewardsBuilder extends TransactionBuilder {
+  protected _stakeAddress: string;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+    this._type = TransactionType.StakingWithdraw;
+  }
+
+  /** @inheritdoc */
+  protected async buildImplementation(): Promise<Transaction> {
+    const tx = await super.buildImplementation();
+    tx.setTransactionType(TransactionType.StakingWithdraw);
+    return tx;
+  }
+
+  /** @inheritdoc */
+  protected fromImplementation(rawTransaction: any): Transaction {
+    return super.fromImplementation(rawTransaction);
+  }
+
+  /** @inheritdoc */
+  protected signImplementation(key: BaseKey): Transaction {
+    return super.signImplementation(key);
+  }
+
+  addWithdrawal(stakingPubKey: string, value: string) {
```

Comment:
why remove network check?

Reply:
i changed input parameter to staking key and value